### PR TITLE
Enhancements to docs and test

### PR DESCRIPTION
**Pull request by KaiVolland**
This adds some docs a test and updates the readme.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -33,11 +45,17 @@ export interface CombinationFilter {
   11?: ComparisonFilter | CombinationFilter | NegationFilter;
 }
 
+/**
+ * A Symbolizer describes the style representation of geographical data.
```
Comment (chrismayer) [inline index.d.ts]: Copy / Paste?

File: index.d.ts [hunk 2]
```diff
@@ -133,16 +172,39 @@ export interface LineSymbolizer extends Symbolizer {
   width?: number;
 }
 
+/**
+ * A Rule aggreagates a specific amount of data and an associated styler.
```
Comment (chrismayer) [inline index.d.ts]: ``amount of data`` = defined / limited by a filter?

``styler`` = Symbolizer?

File: index.d.ts [hunk 3]
```diff
@@ -133,16 +172,39 @@ export interface LineSymbolizer extends Symbolizer {
...
+   *
+   * @param inputStyle
+   */
+  readStyle(inputStyle: any): Promise<Style>;
+
+  /**
+   * Reads the GeoStyler Style and transforms it to the target Style
+   * representation.
+   *
+   * @param style
```
Comment (chrismayer) [inline index.d.ts]: ``style`` => ``geoStylerStyle``

**Comment by KaiVolland** [comment]
Thx for the review. I fixed the noted things.

**Pull request merged**